op-node/p2p/gossip.go

```diff
@@ -469,22 +470,37 @@ func BuildPreconfBlocksResponseValidator(log log.Logger, cfg *rollup.Config, run
 			*res = data[:cap(data)]
 		}
 
-		// message starts with compact-encoding secp256k1 encoded signature
-		signatureBytes, payloadBytes := data[:65], data[65:]
+		// message starts with compact-encoding secp256k1 encoded signature. we can ignore this here.
+		// we dont care who signed the message, we just care that the envelope.Signature is a valid
+		// sequencer originally.
+		payloadBytes := data[65:]
```

Comment:
what is the rationale for adding the signature to the envelope compared to using the existing first 65 bytes of the message to store the sequencer signature instead of the sender signature?

Reply:
salient point - and actually, ithe signing should be entirely removed, because the node gossiping potentially *will not be able to sign* as they dont have a p2p signer ( if they are not a sequencer )

Reply:
Ah I see, because otherwise you can only access the signature in the gossip validator rather than the actual handler which is where you need it. I wonder if an alternative solution could be to make a new type

```go
type SignedExecutionPayloadEnvelope struct {
	Signature ...
	Envelope eth.ExecutionPayloadEnvelope
}
```

and then make the handler `OnUnsafeL2Response` take that from the validator.

Reply:
Is there any advantage to that over just the existing SSZ changes?

Reply:
Ah no, because we would still need to pass it to the driver, it would be a hard fork regardless.

Reply:
Up to you, I think it made sense to use the existing sig field if it already exists, but it's an implementation detail

Reply:
im keeping envelope.Signature to make a clear delineation between the gossiper's signaturea nd the signature of the BlockHash, whichis the only thing signed and persisted ( I think this is OK, please let me know if you think otherwise, but if anything in the block changes the hash should change so its OK we just sign this)